tests/rptest/tests/cloud_storage_usage_test.py

```diff
@@ -84,27 +85,32 @@ def _create_producers(self) -> list[KgoVerifierProducer]:
     def _check_usage(self, timeout_sec):
         bucket_view = BucketView(self.redpanda)
 
+        # The usage inferred from the uploaded manifest
+        # lags behind the actual reported usage. For this reason,
+        # we maintain a sliding window of reported usages and check whether
+        # the manifest inferred usage can be found in it.
+        reported_usage_sliding_window = deque(maxlen=10)
+
         def check():
-            actual_usage = bucket_view.total_cloud_log_size()
+            manifest_usage = bucket_view.total_cloud_log_size()
+
             reported_usage = self.admin.cloud_storage_usage()
+            reported_usage_sliding_window.append(reported_usage)
 
             self.logger.info(
-                f"Expected {actual_usage} bytes of cloud storage usage")
+                f"Expected {manifest_usage} bytes of cloud storage usage")
             self.logger.info(
-                f"Reported {reported_usage} bytes of cloud storage usage")
-            return actual_usage == reported_usage
-
-        # Manifests are not immediately uploaded after they are mutated locally.
-        # For example, during cloud storage housekeeping, the manifest is not uploaded
-        # after the 'start_offset' advances, but after the segments are deleted as well.
-        # If a request lands mid-housekeeping, the results will not be consistent with
-        # what's in the uploaded manifest. For this reason, we wait until the two match.
+                f"Reported usages in sliding window: {reported_usage_sliding_window}"
+            )
+            return manifest_usage in reported_usage_sliding_window
```

Comment:
It's not strictly required that the remote usage is equal to one of the uploaded manifests (we might report usage based on a state in archival_metadata_stm that is quickly superseded and never uploaded  to the bucket as a manifest), but I imagine that in this test we're not uploading aggressively enough for that to happen.